src/Lykke.Service.BlockchainWallets.Services/AddressService.cs

```diff
@@ -82,14 +98,22 @@ public async Task<string> GetUnderlyingAddressAsync(string blockchainType, strin
 
         public async Task<string> GetVirtualAddressAsync(string blockchainType, string address)
         {
-            if (await _capabilitiesService.IsAddressMappingRequiredAsync(blockchainType))
+            var capabilityQueryResult = _blockchainExtensionsService.IsAddressMappingRequired(blockchainType);
+            if (capabilityQueryResult.HasValue && capabilityQueryResult.Value)
             {
                 var apiClient = _blockchainIntegrationService.GetApiClient(blockchainType);
 
-                var virtualAddress = await apiClient.GetVirtualAddressAsync(address);
-                if (!string.IsNullOrWhiteSpace(virtualAddress))
+                try
+                {
+                    var virtualAddress = await apiClient.GetVirtualAddressAsync(address);
+                    if (!string.IsNullOrWhiteSpace(virtualAddress))
+                    {
+                        return virtualAddress;
+                    }
+                }
+                catch (Exception ex)
                 {
```

Comment:
i suggest to use more specif to http error Exception

Reply:
as you log "Unable to access API service for blockchain"

Reply:
Actually, there can be 2 exceptions (in current implementation): `ArgumentException` from `apiClient` because of empty `address` parameter, and some othe exceptions which may be raised internally (including all http-related exceptions).
I added a `when` condition to filter argument validation. But, unfortunately, we do not have (or I can't find) a general http exception in .NET Core like `HttpException` in the classic Framework.